app/code/Magento/Cms/Api/BlockRepositoryInterface.php

```diff
@@ -32,6 +32,16 @@ public function save(Data\BlockInterface $block);
     public function getById($blockId);
 
     /**
+     * Load Block data by given Block identifier
+     *
+     * @param string $blockIdentifier
+     * @param int $storeId
+     * @return \Magento\Cms\Api\Data\BlockInterface
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getByIdentifier($blockIdentifier, $storeId);
```

Comment:
Actually, your PR is redundant, as `Magento\Cms\Api\BlockRepositoryInterface::getById('my-block-identifier')` already works. As is `Magento\Cms\Api\BlockRepositoryInterface::getById(12)`, too. 

The resource model I showed you is just the place where the logic decides which route to go: identifier or numeric id? Again, try it yourself.